packages/viewer/src/components/viewer-icon/viewer-icon.tsx

```diff
@@ -0,0 +1,67 @@
+import { Component, h, Host, Prop } from '@stencil/core';
+import classname from 'classnames';
+
+/**
+ * The names of the icons that the `<vertex-viewer-icon>` element can display.
+ */
+export type ViewerIconName = 'fit-all';
```

Comment:
what are the advantages to a single icon component driven by a property? easier to manage a single file with a big if/switch statement over multiple files?

Reply:
That, and there'll be a lot of boilerplate in Stencil to have separate components for each icon, e.g. each icon would need its own directory + tests.